app/pulp/app/viewsets/task.py

```diff
@@ -0,0 +1,30 @@
+from django_filters.rest_framework import filterset
+
+from pulp.app.models import Task, Worker
+from pulp.app.serializers import TaskSerializer, WorkerSerializer
+from pulp.app.viewsets import NamedModelViewSet
+from pulp.app.viewsets.custom_filters import CharInFilter
+
+
+class TaskFilter(filterset.FilterSet):
+    tags = CharInFilter(name='tag__name', lookup_expr='in')
+
+    class Meta:
+        model = Task
+        fields = ['state', 'worker__name', 'tags']
+
+
+class TaskViewSet(NamedModelViewSet):
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    endpoint_name = 'tasks'
+    filter_class = TaskFilter
+    http_method_names = ['get', 'delete', 'options']
```

Comment:
+1 to not using DELETE to mean cancel. If we still want delete behavior, we'll probably want to create a 'cancel' detail route that works with POST.

http://www.django-rest-framework.org/api-guide/viewsets/#marking-extra-actions-for-routing